spyder_kernels/customize/spyderpdb.py

```diff
@@ -184,7 +190,30 @@ def default(self, line):
                 sys.displayhook = self.displayhook
                 if execute_events:
                      get_ipython().events.trigger('pre_execute')
-                exec(code, ns, locals)
+
+                if uses_comprehension(line):
+                    # Mitigates a cPython bug (https://bugs.python.org/issue41918)
+                    # That prevents running comprehensions with the
+                    # frame locals in pdb.
+                    # See https://bugs.python.org/issue21161
+                    # spyder-ide/spyder#13909.
```

Comment:
```suggestion
                    # There are three potential problems with this approach:
                    # 1. If the code access a globals variable that is
                    #    masked by a locals variable, it will get the locals
                    #   one.
                    # 2. Any edit to that variable will be lost.
                    # 3. The globals will appear to contain all the locals
                    #    variables.
```

- Move the comment below here because its placement was awkward.
- I think it contained three reasons and not two.